app/models/cost_query/cache.rb

```diff
@@ -0,0 +1,75 @@
+#-- copyright
+# OpenProject Reporting Plugin
+#
+# Copyright (C) 2010 - 2014 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#++
+
+module CostQuery::Cache
+  class << self
+
+    def check
```

Comment:
`check` is a very uncommunicative name... For some reason I would even expect it to return a bool, but this is probably just me.

At a minimum level it should probably read `check!`, then I'd at least expect the unexpected... This would be nice, because the only thing it does is to conditionaly call a `!` method.

Ideally the name should be speaking: `try_invalidate!`, `invalidate!`. I admit that I have no clever name at hand that explains the "only if neccessary" part... (`invalidate_if_neccessary!`, anyone?)